packages/std/src/ibc.rs

```diff
@@ -185,11 +187,26 @@ pub struct IbcPacket {
 }
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
-pub struct IbcAcknowledgement {
-    pub acknowledgement: Binary,
+pub struct IbcAcknowledgementWithPacket {
+    pub acknowledgement: IbcAcknowledgement,
     pub original_packet: IbcPacket,
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct IbcAcknowledgement {
+    pub data: Binary,
+    // we may add more info here in the future (meta-data from the acknowledgement)
+    // there have been proposals to extend this type in core ibc for future versions
+}
+
+impl IbcAcknowledgement {
+    pub fn new<T: Serialize>(data: &T) -> StdResult<Self> {
+        Ok(IbcAcknowledgement {
+            data: to_binary(data)?,
+        })
+    }
```

Comment:
I think this connstructor makes too many application specific assumptions. It assumes 1. JSON encoding and 2. the exact JSON format that is derived by the type. It's nice to have that, but let's split it as follows:

```
// More generic, always succeeds
pub fn new<T: Into<Binar>>(data: T) -> Self;

// What you do in the example contract
pub fn encode_json<T: Serialize>(data: &T) -> StdResult<Self>;
```